### PR TITLE
Rename `FlattenStrategy.throttle` to `dropOnBackpressure`.

### PR DESCRIPTION
**Pull request by andersio**
Rename `FlattenStrategy.throttle` to a more common term of art ["drop on backpressure"](https://github.com/ReactiveX/RxJava/wiki/Backpressure-(2.0)#onbackpressuredrop).

#### Checklist
- [x] Updated CHANGELOG.md.

### CODE REVIEW

**Comment by andersio** [comment]
@sergdort @mluisbrown @Marcocanc Thoughts?

**Comment by mluisbrown** [comment]
This might mislead people to believe that RAS supports backpressure like Combine, which it doesn’t.

**Comment by andersio** [comment]
Hmm, it is valid to keep a clear distinction from reactive pull support. Though I am more leaning towards that _backpressure_ is a general scenario that can emerge regardless of the operating model.

**Comment by mluisbrown** [comment]
What’s the motivation to change the name? What’s wrong with .throttle?

**Comment by nkristek** [comment]
In my experience, `throttle` is often mixed up with `debounce` and I see why this naming makes the intention of the operator more clear.

**Comment by mluisbrown** [comment]
> In my experience, throttle is often mixed up with debounce

That is true. However that is in relation to the `throttle(_, on:)` and `debounce(_, on:)` operators, not flattening strategies. But that actually provides a good argument IMO to change the `.throttle` flatten strategy name, to make it distinct from the operator.

Having now seen the RxJava [link](https://github.com/ReactiveX/RxJava/wiki/Backpressure-(2.0)#onbackpressuredrop) that @andersio provided, which provides prior art, I'm inclined to support this name change 👍

File: Sources/Deprecations+Removals.swift [hunk 1]
```diff
@@ -1,6 +1,12 @@
 import Foundation
 import Dispatch
 
+extension FlattenStrategy {
+	/// Renamed to `dropOnBackpressure`.
+	@available(*, deprecated, renamed:"dropOnBackpressure")
```
Comment (Marcocanc) [inline Sources/Deprecations+Removals.swift:6]: Not sure how strict we are on space after colon, but I just noticed this one.

Comment (andersio) [reply Sources/Deprecations+Removals.swift:6]: We've been ad hocly following Swift stdlib style, and there is no space after colon in this context.

**Comment by andersio** [comment]
@mluisbrown 

It was originally proposed as `flatMap(.first)`, but there are concerns about the room of misinterpretation of `first` with `race` and similarly named operators, and not reflecting the "start if no inner is running" aspect of it. So `throttle` was then chosen, as it resembles the semantics of `flatMap(.first)` but in the time domain.

It is only then, when I research for backpressure support in other FRP implementations, I encountered `onBackpressureDrop` as a prior art. IMO it is especially worth adopting because relating to the concept of backpressure is perhaps easier than trying to project `throttle` — an established time domain operation — into another domain.

File: Sources/Flatten.swift [hunk 2]
```diff
@@ -12,7 +12,7 @@ public struct FlattenStrategy {
 		case concurrent(limit: UInt)
 		case latest
 		case race
-		case throttle
+		case dropOnBackpressure
```
Comment (sergdort) [inline Sources/Flatten.swift:15]: `.first` 🙈

Comment (mluisbrown) [reply Sources/Flatten.swift:15]: giphy

Comment (Marcocanc) [reply Sources/Flatten.swift:15]: I prefer `.first` as well

**Pull request closed without merging**